### PR TITLE
Integrate Ghostery Search Extension

### PR DESCRIPTION
**Pull request by chrmod**
Also let fern manage multiple addons.

### CODE REVIEW

File: patches/0007-Remove-bundled-mozilla-extensions.patch [hunk 1]
```diff
@@ -12,14 +12,14 @@ index 974c48c612..0d1e95fa22 100644
 +++ b/browser/extensions/moz.build
 @@ -5,9 +5,6 @@
  # file, You can obtain one at http://mozilla.org/MPL/2.0/.
- 
+
  DIRS += [
 -    'doh-rollout',
 -    'formautofill',
 -    'screenshots',
```
Comment (sammacbeth) [inline patches/0007-Remove-bundled-mozilla-extensions.patch:19]: We could squash this change with patch 3 so all the changes to this file are done in a single commit.

**Review by sammacbeth — COMMENTED** [review]
Looks good! Maybe we can just clean up the patches

File: fern/core/addons.js [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+    const folder = await getPathToCachedAddon(addonName, url);
+    const addonUrlHash = getAddonUrlHash(url);
+    const archive = path.join(cache, `${addonUrlHash}.zip`);
+
+    return [
+      {
+        title: `Download ${addonName}`,
+        skip: () => fileExists(archive),
+        task: () =>
+          promisify(stream.pipeline)(
```
Comment (remusao) [inline fern/core/addons.js:46]: Did you find a way to catch errors here?

File: patches/0007-Remove-bundled-mozilla-extensions.patch (hunk 1 above)
Comment (chrmod) [reply patches/0007-Remove-bundled-mozilla-extensions.patch:19]: Removing a commit form patch list will require to re-name all patches so I would like to do it in separate PR.

File: fern/core/addons.js (hunk 2 above)
Comment (chrmod) [reply fern/core/addons.js:46]: Nope, but the error I was getting was caused by the ghostery-search-extension repository being private. Even if download step do not fail, then the unzip one does, so handing errors here is nice to have only.

**Review by sammacbeth — APPROVED** [review]

File: fern/core/addons.js (hunk 2 above)
Comment (chrmod) [reply fern/core/addons.js]: done

**Pull request merged**